### ISSUE TITLE
npm install fails for node version `16.15.1` and later

### ISSUE THREAD

**Issue by guy-har** [treeverse/lakeFS#3479]
Currently, npm install version `16.15.1` and later fails due to the version of  `react-diff-viewer` 
This resolves by:
- Developers can't run npm install with newer npm versions
- All our workflows are currently fixed on version `16.15.0`

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-diff-viewer@3.1.1
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.0" from the root project
npm ERR!   peer react@">=16.8.0" from @emotion/react@11.9.0
npm ERR!   node_modules/@emotion/react
npm ERR!     peer @emotion/react@"^11.0.0-rc.0" from @emotion/styled@11.8.1
npm ERR!     node_modules/@emotion/styled
npm ERR!       peerOptional @emotion/styled@"^11.3.0" from @mui/material@5.6.4
npm ERR!       node_modules/@mui/material
npm ERR!         @mui/material@"^5.6.4" from the root project
npm ERR!       2 more (@mui/styled-engine, @mui/system)
npm ERR!     peerOptional @emotion/react@"^11.5.0" from @mui/material@5.6.4
npm ERR!     node_modules/@mui/material
npm ERR!       @mui/material@"^5.6.4" from the root project
npm ERR!     2 more (@mui/styled-engine, @mui/system)
npm ERR!   23 more (@emotion/styled, @mui/base, @mui/material, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.3.0 || ^16.0.0" from react-diff-viewer@3.1.1
npm ERR! node_modules/react-diff-viewer
npm ERR!   react-diff-viewer@"^3.1.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^15.3.0 || ^16.0.0" from react-diff-viewer@3.1.1
npm ERR!   node_modules/react-diff-viewer
npm ERR!     react-diff-viewer@"^3.1.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Unfortunately, there is no newer version of `react-diff-viewer`. there is a PR suggesting a fix from Jun 9, 2021, that wasn't merged yet (https://github.com/praneshr/react-diff-viewer/pull/125).

**Comment by johnnyaug** [comment]
Could not reproduce, fixed in #3680

**Comment by johnnyaug** [comment]
Could not reproduce, fixed in #3680
